static/app/views/starfish/components/starfishProjectSelector.tsx

```diff
@@ -12,10 +14,23 @@ import {ALLOWED_PROJECT_IDS_FOR_ORG_SLUG} from 'sentry/views/starfish/allowedPro
 export function StarfishProjectSelector() {
   const {projects, initiallyLoaded: projectsLoaded, fetchError} = useProjects();
   const organization = useOrganization();
-  const {selection, isReady} = usePageFilters();
   const router = useRouter();
+  const {selection} = usePageFilters();
+
+  const allowedProjectIDs: string[] = useMemo(
+    () => ALLOWED_PROJECT_IDS_FOR_ORG_SLUG[organization.slug] ?? [],
+    [organization.slug]
+  );
+
+  const [selectedProjectId, setSelectedProjectId] = useState(
+    selection.projects[0] ?? allowedProjectIDs[0]
+  );
 
-  if (!projectsLoaded || !isReady) {
+  useEffect(() => {
+    setSelectedProjectId(selection.projects[0] ?? allowedProjectIDs[0]);
+  }, [selection.projects, allowedProjectIDs]);
```

Comment:
l: Since this is derived state, can we just update it by [checking prev state directly instead of a `useEffect`](https://react.dev/reference/react/useState#storing-information-from-previous-renders)? It's an unneeded render rn.

Not a huge deal though so feel free to ignore.

Reply:
Great suggestion, thanks Abhi! Updated this to use derived state instead